File: player.py
# implements the Player class for the 10000 dice game

class Player():

    def __init__(self, dice_list):
        self.total_score = 0
        self.on_board = False
        self.dice = dice_list
        self.held_dice = []

    def get_total_score(self):
        return self.total_score
        
    def add_to_score(self, points):
        self.total_score += points
        
    def roll_dice(self):
        if len(self.dice) == 0: #all are held
            tmp = []
            self.dice = self.held_dice
            self.held_dice = tmp
        for d in self.dice:
            d.roll()
            
    def hold(self, dice_to_hold):
        # gets a list of indices, moves them to the held_dice
        for i in dice_to_hold:
            self.held_dice.append(self.dice[i])
        
        # now safely remove held from .dice
        dice_to_hold.reverse()
        for i in dice_to_hold:
            self.dice.pop(i)
        
    def reset_dice(self):
        for d in self.held_dice:
            self.dice.append(d)
        self.held_dice = []
        
    def is_on_board(self, turn_points):
        if not self.on_board: # need 1000 to get on board and score
            if turn_points < 1000: # didn't hit threshold
                print("Not enough to get on the board. Score = 0")
                return False
            else: # score at 1000+
                self.on_board = True
                return True
        else: 
            #already on board
            return True
            

File: java/Player.java
import java.util.*;

public class Player {
    
    int total_score;
    boolean on_board;
    ArrayList<Die> dice_list, held_dice;
    
    public Player() {
        // constructor for the player class
        this.total_score = 0;
        this.on_board = false;
        dice_list = new ArrayList<Die>();
        held_dice = new ArrayList<Die>();
        
        for( int i = 0; i < 6; i++) {
            // populate dice list
            Die d = new Die();
            this.dice_list.add(d); 
        }
    }
    
    public int get_score() {
        return this.total_score;
    }
    
    public void add_to_score(int points) {
        this.total_score += points;
    }
    
    public void roll_dice() {
        Die d;
        for( int i = 0; i < this.dice_list.size(); i++ ){
            d = this.dice_list.get(i);
            d.roll();
        }
    }
    
    public void hold(ArrayList<Integer> dice_to_hold) {
        // move the selected dice from the dice_list to
        // the held_dice list
        for( int i = 0; i < dice_to_hold.size(); i++) {
            int index = dice_to_hold.get(i);
            this.held_dice.add(this.dice_list.get(index));
        }
        
        // now remove the held_dice from the dice_list
        // go backward so there are no index errors
        // check if only one die is selected first
        if (dice_to_hold.size() == 1) {
            this.dice_list.remove(dice_to_hold.get(0));
        }
        else {   // more than one die
            for(int i = dice_to_hold.size(); i > 0; i--) {
                int index = dice_to_hold.get(i - 1);
                this.dice_list.remove(index);
            }
        }
        // might have to add "trimToSize()" on the dice_list
    }
    public void reset_dice() {
        this.dice_list.clear();
        for( int i = 0; i < 6; i++) {
            // populate dice list
            Die d = new Die();
            this.dice_list.add(d); 
        }
        this.held_dice.clear();
    }
    
    public boolean is_on_board(int turn_points) {
        if (!this.on_board) {
            if (turn_points < 1000) {
                // didn't hit threshold
                System.out.println("Not enough to get on the board. Score = 0");
                return false;
            }
            else { // score at 1000+
                this.on_board = true;
                return true;
            }
        }
        else { // already on board
            return true;
        }
    }
    
    public ArrayList<Die> get_rolled_dice(){
        return this.dice_list;
    }
    
    
    public static void main(String[] args) {
        Player p = new Player();
        System.out.println(p.get_score());
        
        for(int i = 0; i < 4; i++) {
            p.roll_dice();
            System.out.println();
        }
        
    }
    
    
}